### PR TITLE
Add function to extract values from FluxMaps

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
**Description**
This pull request is an initial function as the first approach for Issue #4470.
The function `find_peaks_in_flux_map` has been added to `gammapy.estimators.utils`, which utilises the methods from `find_peaks`.
The column names for the table are chosen to match the `FluxPointsEstimator` table column names. If anyone has better naming ideas, please let me know.

### CODE REVIEW

File: gammapy/estimators/utils.py [hunk 1]
```diff
@@ -116,6 +118,99 @@ def find_peaks(image, threshold, min_distance=1):
...
+    # Remove non-finite values to avoid warnings or spurious detection
+    data = image.sum_over_axes(keepdims=False).data
+    data[~np.isfinite(data)] = np.nanmin(data)
+
+    # Handle edge case of constant data; treat as no peak
+    if np.all(data == data.flat[0]):
+        return Table()
+
+    # Run peak finder
+    data_max = scipy.ndimage.maximum_filter(data, size=size, mode="constant")
```
Comment (adonath) [inline gammapy/estimators/utils.py:170]: Why not just rely on `find_peaks`? You can later use the pixel positions to extract the values from the `npred` and excess maps. And just extend the returned table.

File: gammapy/estimators/utils.py [hunk 2]
```diff
@@ -116,6 +118,99 @@ def find_peaks(image, threshold, min_distance=1):
...
+    flux_peaks = flux_data[0, y, x]
+    flux_err_data = maps["flux_err"].data
+    flux_err_peaks = flux_err_data[0, y, x]
+
+    table = Table()
+    table["x"] = x
+    table["y"] = y
+    table["ra"] = coord.ra
+    table["dec"] = coord.dec
+    table["sqrt_ts"] = sqrt_ts_values * image.unit
```
Comment (adonath) [inline gammapy/estimators/utils.py:194]: Is it possible to replace this by a loop over all `FluxMap` quantities? Note that some quantities are optional, but those would only be added if available.

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks a lot @Astro-Kirsty, this is a very useful feature! I have left some comment to simplify the implementation and make more use of already existing functionality.

File: gammapy/estimators/utils.py (hunk 2 above)
Comment (Astro-Kirsty) [reply gammapy/estimators/utils.py]: I created a loop over the `available_quantities` as well as adding in the `flux` and `flux_err`. I guess the question is whether we want to loop over `all_quantities` or leave it as is?

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . 

If you want to extract all information from the `FluxMaps` object you can use the `get_flux_points()` method. See https://github.com/gammapy/gammapy/blob/e45cb44c8bfb6b3029e134c571ca0bdacb76cbfb/gammapy/estimators/map/core.py#L720
Interestingly, the results of your function could be also returned in a list of `FluxPoints`...

File: gammapy/estimators/utils.py [hunk 3]
```diff
@@ -116,6 +118,80 @@ def find_peaks(image, threshold, min_distance=1):
...
 
+def find_peaks_in_flux_map(maps, threshold, min_distance=1):
+    """Find local TS peaks for a given Map.
+
+    Utilises the find_peaks function to find various parameters from FluxMaps.
+
+    Parameters
+    ----------
+    maps : `~gammapy.estimators.map.FluxMaps`
+    threshold : float or array-like
```
Comment (bkhelifi) [inline gammapy/estimators/utils.py:129]: The docstring does not state is the threshold is in TS... can you precise it?

File: gammapy/estimators/utils.py [hunk 4]
```diff
@@ -116,6 +118,80 @@ def find_peaks(image, threshold, min_distance=1):
...
 
 
+def find_peaks_in_flux_map(maps, threshold, min_distance=1):
+    """Find local TS peaks for a given Map.
+
+    Utilises the find_peaks function to find various parameters from FluxMaps.
+
+    Parameters
+    ----------
+    maps : `~gammapy.estimators.map.FluxMaps`
```
[the remark this answers is not in the document]
Comment (Astro-Kirsty) [reply gammapy/estimators/utils.py:128]: Thanks @bkhelifi. 
It is a FluxMaps object which contains multiple quantities (i.e maps). Happy to change to be 'map' if you think it is confusing for the user

File: gammapy/estimators/utils.py [hunk 5]
```diff
@@ -116,6 +118,80 @@ def find_peaks(image, threshold, min_distance=1):
...
+        peaks = values[0, y, x]
+        table[name] = peaks
+
+    flux_data = maps["flux"].quantity
+    table["flux"] = flux_data[0, y, x]
+    flux_err_data = maps["flux_err"].quantity
+    table["flux_err"] = flux_err_data[0, y, x]
+
+    for column in table.colnames:
+        if column.startswith(("flux", "flux_err")):
```
Comment (registerrier) [inline gammapy/estimators/utils.py:173]: Is there a specific reason for defining a different format for the flux columns?

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . This looks good. I have left some small inline suggestions.

File: gammapy/estimators/utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Astro-Kirsty) [reply gammapy/estimators/utils.py:128]: Thanks @registerrier, I have implemented those changes

File: gammapy/estimators/utils.py (hunk 5 above)
Comment (Astro-Kirsty) [reply gammapy/estimators/utils.py:176]: For readability of the flux values -- typical fluxes of 1e-13, for example

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . No further comments.

**Comment by registerrier** [comment]
CI fail is unrelated. Merging.

**Pull request merged**